src/id/shinigami/src/eu/kanade/tachiyomi/extension/id/shinigami/Shinigami.kt

```diff
@@ -83,70 +56,195 @@ class Shinigami : Madara("Shinigami", "https://shinigami09.com", "id"), Configur
         .rateLimit(3)
         .build()
 
-    // Tags are useless as they are just SEO keywords.
-    override val mangaDetailsSelectorTag = ""
+    override fun headersBuilder(): Headers.Builder = Headers.Builder()
+        .add("X-Requested-With", randomString((1..20).random())) // added for webview, and removed in interceptor for normal use
 
-    override val chapterUrlSelector = "div.chapter-link:not([style~=display:\\snone]) a"
+    private fun randomString(length: Int): String {
+        val charPool = ('a'..'z') + ('A'..'Z')
+        return List(length) { charPool.random() }.joinToString("")
+    }
```

Comment:
More performant and succinct:
```suggestion
    private fun randomString(length: Int) = buildString {
        val charPool = ('a'..'z') + ('A'..'Z')
        repeat(length) { append(charPool.random()) }
    }
```